src/main/java/de/ipvs/as/mbp/service/UnitConverterService.java

```diff
@@ -1,11 +1,13 @@
 package de.ipvs.as.mbp.service;
 
 import de.ipvs.as.mbp.domain.valueLog.ValueLog;
+import org.bson.Document;
 import org.springframework.stereotype.Component;
 
 import javax.measure.converter.UnitConverter;
 
 /**
+ * TODO Must be updated if it should be used with complex iot data
```

Comment:
If I am not wrong this service does still work with the monitoring operators (which aren't allowed to have a complex data model). So it it does work and is still in use but not for complex data. A main problem I see for complex data is that we have no UI for the use case of changing units of complex data structures. 

To enable something like a unit converter for complex data it would be necessary to specify the path to the json field (only leaf nodes are relevant here because only they can have units) and the wanted unit. Then one could find the corresponding data model leaf node by iterating over the "leafNodes"-List inside the DataModelTree. With the DataModelTree one can use the "DocumentReader" class from the Utils package to read and overwrite the Document field of a ValueLog at the respective place. However for arrays it would be a little bit more complex, because every array index combination for multidimensional arrays (arr[2][3] has 6 entries) has to be considered individually. So it would be possible but is not that trivial (one would have to be willing to build a UI for it as well to make it worthwhile).